[Kiat Lun] IP

KiatLun opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

KiatLun committed:
Level-0

KiatLun committed:
Level 1. Echo

KiatLun committed:
Level 2. Add, List

KiatLun committed:
Level 3. Mark as Done

KiatLun committed:
Level 4. Todos, Events, Deadlines

KiatLun committed:
Level 5. Handle Errors

KiatLun committed:
Level 6. Delete

KiatLun committed:
Level 7. Save

KiatLun committed:
A-TextUiTesting

KiatLun committed:
branch Level 7. Save

KiatLun committed:
Merge branch 'branch-level-7'

nigel27022001 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+public class Task {
+    private String done;
```
Should the indicator of whether a Task is done be a boolean?
```suggestion
    private boolean isDone;
```

nigel27022001 commented on src/main/java/TaskHandler.java:
```diff
@@ -0,0 +1,186 @@
+                            "What tasks do you want to unmark?");
+                }
+
+                int number = Integer.parseInt(temp[1]) ;
```
Is there any exception you should consider when trying to parse a char?
```java
int number = Integer.parseInt(temp[1]);
```

nigel27022001 commented on src/main/java/TaskHandler.java:
```diff
@@ -0,0 +1,186 @@
+
+public class TaskHandler{
+
+    public static void addTasks(String input, ArrayList<Task> storage, Path filePath) throws IOException {
```
You can try to separate the "adding task" component and "command" components. eg. "command" components include marking and unmarking tasks. This is because this addTasks method is quite long.

nigel27022001 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,12 @@
+public class ToDo extends Task {
+    String label = "[T]";
+
+    public ToDo(String msg) {
+        super(msg);
+    }
+    @Override
+    public String toString() {
+        return label + super.toString() + " ";
+    }
+
+}
```
I like how you did the Todo,Event and Deadline. It is clean and concise.

nigel27022001 commented on src/main/java/FileHandler.java:
```diff
@@ -0,0 +1,48 @@
+    public static boolean isMarked(String input) {
+        return String.valueOf(input.charAt(4)).equals("X");
+    }
+    public static Path addTasks(ArrayList<Task> storage, Path filePath) throws IOException {
```
Should the method be named addTasks? because it might be confusing as it has a similar method in TaskHandler

nigel27022001 reviewed (commented):
Good Job so far! Looks like you're quite busy and haven't caught up but keep pushing on. Do take note of some of the comments I've made.

KiatLun committed:
Level 8. Date and Times

KiatLun committed:
Merge branch 'branch-Level-8'

sun-ruiheng commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+    @Override
+    public boolean inRange(LocalDate fromDate, LocalDate toDate) {
+        return (date.isBefore(toDate) && date.isAfter(fromDate))
+                || date.equals(fromDate) || date.equals(toDate);
```
Good indentation!

sun-ruiheng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,51 @@
+        ArrayList<Task> storage = new ArrayList<>();
+        Path filePath = Paths.get("./data/duke.txt");
+        Path writeToFile = FileHandler.addTasks(storage, filePath);
+        Scanner scan = new Scanner(System.in);
```
Scanner object should be named as a noun, like scanner, or sc.
"scan" is a verb and sounds like an action, i.e. a _method_.

sun-ruiheng commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+    
+
+    @Override
+    public boolean inRange(LocalDate startDate, LocalDate endDate) {
```
would be clearer if it's named `isInRange` rather than `inRange`

sun-ruiheng commented on src/main/java/TaskHandler.java:
```diff
@@ -0,0 +1,232 @@
+            }
+
+        }
+
```
Can be more consistent with number of line spacings 🤔

sun-ruiheng commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,14 @@
+import java.time.LocalDate;
```
Nice OOP by segregating the different types of tasks!

sun-ruiheng commented on src/main/java/TaskHandler.java:
```diff
@@ -0,0 +1,232 @@
+                String[] test = input.split(" ");
+                if (test.length <= 1) {
+                    throw new DukeException("__________________________________\n" +
+                            "So where is the deadline task????");
```
Quite a funny reply.
Nice catch to ensure input makes sense, and to ensure future codes don't fail

sun-ruiheng reviewed (commented):
Looks pretty good overall, just a few styling nits here and there! Well done! 💪

KiatLun committed:
Use More OOP

KiatLun committed:
Organise into Duke package

KiatLun committed:
Merge branch 'add-gradle-support'

KiatLun committed:
Use Gradle

KiatLun committed:
Add JUnit Tests

KiatLun committed:
Use Gradle for JUnit

KiatLun committed:
Added JavaDoc

KiatLun committed:
Merge branch 'branch-A-JavaDoc'

KiatLun committed:
Follow Coding Standard

KiatLun committed:
Merge branch 'branch-A-CodingStandard'

KiatLun committed:
Level 9. Find

KiatLun committed:
Merge branch 'branch-Level-9'

KiatLun committed:
Level 10. GUI

KiatLun committed:
Level 10. GUI (part 2)

KiatLun committed:
Merge branch 'branch-Level-10'

KiatLun committed:
Implemented Category C extention: C-DetectDuplicates

Detect duplicates in the stored tasks list. If there is already a task exactly
same as the one to be added, the program will not add task to list and prompt
user to write new tasks.

Implemented isDuplicate(Task) in the 3 types of task class to check if task is the same
as the one specified.

Implemented hasDuplicates(Task, TaskList) to check for duplicates in the Tasklist.

KiatLun committed:
Merge branch 'branch-BCD-extension'

KiatLun committed:
Improving code quality in TaskHandler class

Previously for the method checkSchedule(String, String, TaskList) there was
a problem of deep nesting. I fixed it by adding a new method checkInRange(TaskList, LocalDate, LocalDate)
in TaskList which loops through the TaskList to find task within the date range.

KiatLun committed:
A-Assertions

KiatLun committed:
Merge branch 'branch-A-Assertions'

KiatLun committed:
Added User Guide

KiatLun committed:
Worked on improving code quality

KiatLun committed:
Merge branch 'branch-A-CodeQuality'

KiatLun committed:
A-BetterGui

KiatLun committed:
Update README file

KiatLun committed:
Additional modifications

KiatLun committed:
Modify README file and Ui file

KiatLun committed:
change to Ui.png

KiatLun committed:
Made new packages and implemented the Parser Class to parse user inputs and separated each command to their separate command class

KiatLun committed:
Merge pull request #3 from KiatLun/Branch-Final-Update
